command/agent/config.go

```diff
@@ -1089,6 +1093,57 @@ func (n *NormalizedAddrs) Copy() *NormalizedAddrs {
 	return &nn
 }
 
+// SocketConfig contains the path and configuration for a unix domain socket
+// listener.
+type SocketConfig struct {
+	Path  string `hcl:"path"`
+	User  string `hcl:"user"`
+	Mode  string `hcl:"mode"`
+	Group string `hcl:"group"`
+}
+
+func (s *SocketConfig) Copy() *SocketConfig {
+	if s == nil {
+		return nil
+	}
+	ns := *s
+	return &ns
+}
+
+// Merge merges two SocketConfigs together, preferring values from the argument.
+func (s *SocketConfig) Merge(b *SocketConfig) *SocketConfig {
+	if b == nil {
+		return s
+	}
+	var result SocketConfig
+	if s != nil {
+		result = *s
+	}
+	if b.Path != "" {
+		result.Path = b.Path
+	}
+	if b.User != "" {
+		result.User = b.User
+	}
+	if b.Group != "" {
+		result.Group = b.Group
+	}
+	if b.Mode != "" {
+		result.Mode = b.Mode
+	}
```

Comment:
These can be 1-liners with https://github.com/hashicorp/nomad/blob/main/helper/funcs.go#L492-L500